openapi.yml

```diff
@@ -0,0 +1,576 @@
+        - name: thread_id
+          in: path
+          description: UUID of the thread to patch
+          required: true
+          schema:
+            type: string
+            format: uuid
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Thread'
+        '400':
+          description: Client error
+        '404':
+          description: Thread not found
+        '405':
+          description: Method not allowed
+    delete:
+      tags:
+        - Threads
+        - New
+      summary: Delete a thread by the thread_id
+      description: Delete a thread by the `thread_id`
+      parameters:
+        - name: thread_id
+          in: path
+          description: UUID of the thread to be deleted
+          required: true
+          schema:
+            type: string
+            format: uuid
+      responses:
+        '204':
+          description: Successfully deleted
+        '404':
+          description: Thread not found
+        '405':
+          description: Method not allowed
+    put:
+      tags:
+        - Threads
+        - Existing
+      summary: Put a thread by the thread id
+      description: Update the thread's read status by the `thread_id`. This will only update the `is_read_by_respondent` **OR** `is_read_by_internal` depending on who is reading the message, the other should already be set to read since they have sent the message?
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                is_read_by_respondent:
+                  type: boolean
+                  example: true
+                is_read_by_internal:
+                  type: boolean
+                  example: true
+            example:
+              is_read_by_respondent: true
+              is_read_by_internal: true
+      parameters:
+        - name: thread_id
+          in: path
+          description: UUID of the thread to update
+          required: true
+          schema:
+            type: string
+            format: uuid
+      responses:
+        200:
+          description: Successfully set the thread as [read|unread] for [internal user|respondent]
+        400:
+          description: Invalid thread id
+        404:
+          description: Thread not found
+        405:
+          description: Method not allowed
+  /threads/count:
+    get:
+      tags:
+        - Threads
+        - Existing
+      summary: Get count of threads for a user
+      description: Return count of threads for a user depending on a number of filters
+      parameters:
+        - name: user_id
+          in: query
+          description: User UUID assigned to the theads
+          schema:
+            type: string
+            format: uuid
+            example: 6934f0c7-b595-435b-aa8f-291d6b6070d3
```

Comment:
Depending on decision to continue to use the JWT (frontstage `role: respondent` rops `role: internal`) this parameter may be redundant.

Reply:
Decision taken to continue to use JWT authentication